core/embed/rust/src/ui/model_tt/component/pin.rs

```diff
@@ -111,7 +103,7 @@ impl PinDialog {
                 i + 1 + 3
             });
             let text: &[u8; 1] = digits[i];
-            Child::new(Button::with_text(area, text, theme::button_default()))
+            Child::new(Button::with_text(area, text))
```

Comment:
I'm not sure about into_child honestly. What do you think about it?

Reply:
It's used in a lot of places.

I don't have an opinion either way but let's be consistent. Are there any drawbacks / benefits of `into_child()` vs `Child::new` ?

Reply:
It's a bit nicer when you're chaining more calls.

Reply:
I reviewed the usages, and it seems `into_child()` is exclusively used when constructing layouts. Given that the `ComponentExt` is weird (i now understand why you weren't sure about it), i propose removing it and using the following convention instead of `into_child()`:
```rust
Self {
    content: Child::new(content),
    left: Child::new(left),
    ...
}
```

Reply:
> Given that the ComponentExt is weird...

`ComponentExt` itself is alright I think, why do you find it weird?

Reply:
well, its only purpose seems to be injecting `into_child()` into a `Component` through trickery (because Rust won't allow us to do it directly) -- but for this trickery to work, we need to explicitly use `ComponentExt`, which does not at all sound like something I would look for.

If ComponentExt provided some sort of wider extended functionality, I wouldn't think twice, but it doesn't seem warranted for a method that we don't really need all that much.

Reply:
Oh, OK. Extension traits are fairly common practice in Rust (even std), and I think in the future, we would use the trait for other component sugar, but you're right that this specific method doesn't warrant it very well.

Reply:
Oh and it's definitely not a trickery :)

Reply:
Anyway, I would vote for keeping it, as I think it'll get more useful in the future.

Reply:
in the end i'm leaving this as-is

it still vexes me that there's no "one obvious way to do it" though